### PR TITLE
Use async api

### PR DESCRIPTION
**Pull request by schabannes**
Requires PR #194 
Use async API in IGraphQLExecutor and IRequestHandler.
Remove sync conterpart methods.

### CODE REVIEW

File: src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs [hunk 1]
```diff
@@ -261,12 +260,9 @@ public GraphQLExecutor NewExecutor(IRequestDeserializer requestDeserializer = nu
             return this;
         }
 
-        public string SerializeResult(ExecutionResult result)
-        {
-            return _documentWriter.Write(result);
-        }
+        public async Task<string> SerializeResultAsync(ExecutionResult result) => await _documentWriter.WriteToStringAsync(result);
```
Comment (sungam3r) [inline src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs]: Async/await can be removed

File: src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs [hunk 2]
```diff
@@ -329,10 +330,12 @@ public string SerializeResult(ExecutionResult result)
             return result;
         }
 
-        internal IValidationResult Validate(string queryString)
+        internal async Task<IValidationResult> ValidateAsync(string queryString)
         {
             var document = _documentBuilder.Build(queryString);
-            return _documentValidator.Validate(queryString, _schema, document);
+            var result = _documentValidator.Validate(queryString, _schema, document);
```
Comment (sungam3r) [inline src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs]: Validate will be async soon too.

File: src/GraphQL.Conventions/Adapters/Engine/GraphQLExecutor.cs [hunk 3]
```diff
@@ -145,6 +145,6 @@ await _engine
                     listeners: _documentExecutionListeners)
                 .ConfigureAwait(false);
 
-        public IValidationResult Validate() => _engine.Validate(_queryString);
+        public async Task<IValidationResult> ValidateAsync() => await _engine.ValidateAsync(_queryString);
```
Comment (sungam3r) [inline src/GraphQL.Conventions/Adapters/Engine/GraphQLExecutor.cs]: Async await not needed

**Comment by schabannes** [comment]
Included in feature/3.0-preview

**Pull request closed without merging**